### ISSUE TITLE
Doesn't work with lombok

### ISSUE THREAD

**Issue by RaviH** [ngs-doo/dsl-json#145]
Please try to run the Example in Jackson-example module: https://github.com/RaviH/dsl-json

I get the error below:

```
Exception in thread "main" com.dslplatform.json.ConfigurationException: Unable to serialize provided object. Failed to find serializer for: class com.dslplatform.jackson.Example$Model
	at com.dslplatform.json.DslJson.serialize(DslJson.java:2850)
	at com.dslplatform.jackson.Example.main(Example.java:184)

Process finished with exit code 1
```

Does dsl-json not work with lombok?

**Comment by zapov** [comment]
I tried it a while ago and couldn't make it work since Lombok uses some special compilation process and avoids builtin java code generation tools. As I'm don't really know much about it I didn't try to look if there is a way to call Java annotation processor after Lombok finishes so I just created an example for AutoValue which respects the Java annotation processor compilation step.

**Comment by RaviH** [comment]
Hey @zapov 

Thank you for getting back to me. We use `map struct`, `lombok`, `micronaut` which all work with annotation processor path.  I would love to use dsl-json but it would need to work with all of the above.

Do you think it's possible to work with `lombok` folks to get dsl-json to work with it?

**Comment by RaviH** [comment]
Also, this gives me a compile time error:

```
public class StatusDefinition {

    Integer id;
    String name;
    Long lastModified;
    String statusDescription;

    @JsonCreator
    public StatusDefinition(
            @JsonProperty("id")
                    Integer id,
            @JsonProperty("statusdefinitionid")
                    Integer ncsId,
            @JsonProperty("name")
                    String name,
            @JsonProperty("stausdefinitionname")
                    //Misspelled on purpose
                    String ncsName,
            @JsonProperty("lastModified")
                    Long lastModified,
            @JsonProperty("statusDescription")
                    String statusDescription
    ) {

        this.id = id.defaultIfNull(ncsId);
        this.name = name.defaultIfNullOrBlank(ncsName);
        this.lastModified = lastModified;
        this.statusDescription = statusDescription;
    }

    /**
     * Needed because @AllArgsConstructor causes object mapping issues due to constructor conflicts
     */
    public StatusDefinition() {

    }

    public Integer getId() {

        return this.id;
    }

    public String getName() {

        return this.name;
    }

    public Long getLastModified() {

        return this.lastModified;
    }

    public String getStatusDescription() {

        return this.statusDescription;
    }

    public StatusDefinition setId(Integer id) {

        this.id = id;
        return this;
    }

    public StatusDefinition setName(String name) {

        this.name = name;
        return this;
    }

    public StatusDefinition setLastModified(Long lastModified) {

        this.lastModified = lastModified;
        return this;
    }

    public StatusDefinition setStatusDescription(String statusDescription) {

        this.statusDescription = statusDescription;
        return this;
    }

    public boolean equals(final Object o) {

        if (o == this) {
            return true;
        }
        if (!( o instanceof StatusDefinition )) {
            return false;
        }
        final StatusDefinition other = (StatusDefinition) o;
        if (!other.canEqual((Object) this)) {
            return false;
        }
        final Object this$id = this.getId();
        final Object other$id = other.getId();
        if (this$id == null ? other$id != null : !this$id.equals(other$id)) {
            return false;
        }
        final Object this$name = this.getName();
        final Object other$name = other.getName();
        if (this$name == null ? other$name != null : !this$name.equals(other$name)) {
            return false;
        }
        final Object this$lastModified = this.getLastModified();
        final Object other$lastModified = other.getLastModified();
        if (this$lastModified == null ? other$lastModified != null : !this$lastModified.equals(other$lastModified)) {
            return false;
        }
        final Object this$statusDescription = this.getStatusDescription();
        final Object other$statusDescription = other.getStatusDescription();
        if (this$statusDescription == null ? other$statusDescription != null : !this$statusDescription.equals(
                other$statusDescription)) {
            return false;
        }
        return true;
    }

    protected boolean canEqual(final Object other) {

        return other instanceof StatusDefinition;
    }

    public int hashCode() {

        final int PRIME = 59;
        int result = 1;
        final Object $id = this.getId();
        result = result * PRIME + ( $id == null ? 43 : $id.hashCode() );
        final Object $name = this.getName();
        result = result * PRIME + ( $name == null ? 43 : $name.hashCode() );
        final Object $lastModified = this.getLastModified();
        result = result * PRIME + ( $lastModified == null ? 43 : $lastModified.hashCode() );
        final Object $statusDescription = this.getStatusDescription();
        result = result * PRIME + ( $statusDescription == null ? 43 : $statusDescription.hashCode() );
        return result;
    }

    public String toString() {

        return "StatusDefinition(id=" + this.getId() + ", name=" + this.getName() + ", lastModified=" + this.getLastModified() + ", statusDescription=" + this
                .getStatusDescription() + ")";
    }
}
```
```
Error:(13, 5) java: Unable to find matching property: 'ncsId' used in constructor. Either use annotation processor on source code, on bytecode with -parameters flag (to enable parameter names) or manually create an instance via converter
```

**Comment by zapov** [comment]
I looked again at Lombok and made a simple example with it here: https://github.com/ngs-doo/dsl-json/tree/master/examples/Lombok

As far as the ncsId problem goes, this currently does not work since dsl-json always tries to pair in/out parameters. There is no out parameter for this so it throws that strange error message.
Ideally there should be additional parameters to specify if you only have input-only or output-only parameters, but this is something which needs to be improved/supported.

**Comment by sweintritt** [comment]
I had two problems in my project using lombok and dsl-json.

First I had to declare the lombok annotation processor before dsl-json in my gradle build to make it work

```
annotationProcessor "org.projectlombok:lombok:1.18.8"
annotationProcessor "com.dslplatform:dsl-json-java8:1.9.3"
```

Another problem I had was that booleans where not contained in the serialized JSON. Problem is DSL-JSON is searching for getXXX and setXXX to access fields but lombok generates isXXX methods for booleans. After I implemented the getters myself the serialization worked fine. An alternative would be to use Boolean instead of boolean because lombok will then generate getXXX methods.
